src/main/java/org/codehaus/mojo/versions/ordering/MajorMinorIncrementalFilter.java

```diff
@@ -51,11 +51,11 @@
 public class MajorMinorIncrementalFilter
 {
 
-    private boolean allowMajorUpdates;
+    private final boolean allowMajorUpdates;
```

Comment:
in other places you remove the final, here you add one. I would like to keep the final in other places, too, so you actually don't overwrite the vars (helps the compiler a bit).

Reply:
final for class fields makes sense. after the constructor has completed, any final fields are guaranteed to be visible in other threads immediately.  But I agree this is not mandatory here as it is not a multi threaded env.
But frankly final for local variable available 2 lines is just useless...
If you can prove such paradigm makes sense I will be happy to hear:
```
        catch ( final InterruptedException ie )
        {
            throw new MyException(ie);
        }
```
it just makes checkstyle happy....